apps/api/src/app/admin/admin.service.ts

```diff
@@ -295,6 +298,27 @@ export class AdminService {
     dataSource,
     symbol
   }: UniqueAsset): Promise<AdminMarketDataDetails> {
+    let activitiesCount: EnhancedSymbolProfile['activitiesCount'] = 0;
+    let dateOfFirstActivity: EnhancedSymbolProfile['dateOfFirstActivity'];
+    let currency: EnhancedSymbolProfile['currency'] = '-';
+
+    if (isCurrency(symbol.replace(DEFAULT_CURRENCY, ''))) {
+      currency = symbol.replace(DEFAULT_CURRENCY, '');
+      const { _count, _min } = await this.prismaService.order.aggregate({
```

Comment:
Can you extract this as a function, move it to `OrderService` and reuse it also in `getMarketDataForCurrencies()`?